### PR TITLE
Site Editor: Change the `<` link to the theme showcase 

### PR DESCRIPTION
**Pull request by okmttdhr**
## Proposed changes:

* Change the Site Editor's `<` link to the theme showcase page. See https://github.com/Automattic/wp-calypso/issues/80595.
* This PR works with https://github.com/WordPress/gutenberg/pull/54242 while this does not break anything without https://github.com/WordPress/gutenberg/pull/54242.

- [x] I'll create a follow-up PR to move this into `jetpack-mu-wpcom` https://github.com/Automattic/jetpack/pull/32913#discussion_r1318140030
	- -> https://github.com/Automattic/jetpack/pull/33024

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Apply https://github.com/WordPress/gutenberg/pull/54242 to your site (because this PR works with https://github.com/WordPress/gutenberg/pull/54242 while this does not break anything without it)
	* On Simple sites
		* Sync the change to your sandbox with AWE or rsync PCYsg-t5D-p2
	* On Atomic sites
		* Uninstall gutenberg
		* Checkout to https://github.com/WordPress/gutenberg/pull/54242
		* npm run dev
		* Create a new directory with including only `build/***`, `gutenberg.php`, `lib/***`, `packages/***`, `post-content.php`, `vendor/***`, and zip it
		* Upload the zip to `~/htdocs/wp-content/plugins` on your site 
		* Run `unzip gutenberg.zip` on your site and activate it
* Apply this patch to your site by doing https://github.com/Automattic/jetpack/pull/32913#issuecomment-1709419749
	* Note that you must set `define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true );` to your wp-config.php on Atomic sites.
* Go to the Theme Showcase 
* Click the Live Preview option on the three-dots menu on the Theme Card
* Confirm if the `<` link point to the Theme Showcase page

Screen Shot 2023-09-07 at 15 05 41

## Related

- https://github.com/Automattic/jetpack/pull/27601
- https://github.com/Automattic/wp-calypso/pull/81387

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php [hunk 1]
```diff
@@ -48,6 +48,23 @@ public function reregister_menu_items() {
 		ksort( $GLOBALS['menu'] );
 	}
 
+	/**
+	 * Returns the Calypso origin.
+	 *
+	 * @return string
+	 */
+	private function get_calypso_origin() {
```
Comment (arthur791004) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: It's related to the Calypso, I think we can keep using `https://wordpress.com` for now and continue to migrate to jetpack-mu-wpcom plugin in the follow-up PR. WDYT?

Comment (okmttdhr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: @arthur791004 Do you mean that we should move both `site_editor_dashboard_link` and `site_editor_theme_preview_back_link` into `jetpack-mu-wpcom` as well as `get_calypso_origin` eventually? 👀

Comment (arthur791004) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: YES!

Comment (okmttdhr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: That makes sense then 🙂 Let me fix it!

Comment (okmttdhr) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: I addressed it and added a PR description.

**Review by arthur791004 — DISMISSED** [review]
LGTM 🎉

**Review by jeherve — DISMISSED** [review]

**Comment by okmttdhr** [comment]
Hello, @arthur791004 @jeherve. I made a change https://github.com/Automattic/jetpack/pull/32913/commits/9993abf5b729fb942d42bfca20452840c651636b to align with [the Gutenberg implementation](https://github.com/WordPress/gutenberg/pull/54242). Could you take a look and ✅ again? Thanks! 🙏

**Review by arthur791004 — APPROVED** [review]

**Pull request merged**